### PR TITLE
refactor logging

### PR DESCRIPTION
**Pull request by waynz0r**
## Description

change logging strategy to use pr_debug for most places and use dynamic debug if needed

## Type of Change

- [x] Refactor

### CODE REVIEW

**Review by bonifaido — CHANGES_REQUESTED** [review]
I would add some instructions to the README how to extract those logs on lower log levels.

File: csr.c [hunk 1]
```diff
@@ -101,19 +103,19 @@ csr_result csr_gen(csr_module *csr, i32 priv_key_buff_ptr, i32 priv_key_buff_len
...
+    if (parameters->field)                                                           \
+    {                                                                                \
+        field##_len = strlen(parameters->field);                                     \
+        field##_ptr = alloc_and_copy_parameter(parameters->field, field##_len, csr); \
+        if (field##_ptr == -1)                                                       \
+        {                                                                            \
+            result.err = "error during allocating ptr with length for " #field;      \
+            goto bail;                                                               \
+        }                                                                            \
+    }
```
Comment (baluchicken) [inline csr.c:118]:
```suggestion
    } \
```

File: device_driver.c [hunk 2]
```diff
@@ -301,19 +303,19 @@ static int parse_command(const char *data)
...
         const char *command = json_object_get_string(root, "command");
 
-        pr_info("nasp: command %s", command);
+        pr_debug("incoming command # command[%s]", command);
 
         if (strcmp("load", command) == 0)
         {
             const char *name = json_object_get_string(root, "name");
-            pr_info("nasp: loading module: %s", name);
+            pr_info("load module # name[%s]", name);
```
Comment (baluchicken) [inline device_driver.c:311]:
```suggestion
            pr_debug("load module # name[%s]", name);
```

File: device_driver.c [hunk 3]
```diff
@@ -322,14 +324,14 @@ static int parse_command(const char *data)
             const char *entrypoint = json_object_get_string(root, "entrypoint");
             if (entrypoint == NULL)
             {
-                pr_info("nasp: setting default module entrypoint \"%s\"", DEFAULT_MODULE_ENTRYPOINT);
+                pr_info("setting default module entrypoint # entrypoint[%s]", DEFAULT_MODULE_ENTRYPOINT);
```
Comment (baluchicken) [inline device_driver.c:327]:
```suggestion
                pr_debug("setting default module entrypoint # entrypoint[%s]", DEFAULT_MODULE_ENTRYPOINT);
```

File: device_driver.c [hunk 4]
```diff
@@ -339,63 +341,65 @@ static int parse_command(const char *data)
         }
         if (strcmp("reset", command) == 0)
         {
-            pr_info("nasp: reseting vm");
+            pr_info("reseting vm");
```
Comment (baluchicken) [inline device_driver.c:344]:
```suggestion
            pr_debug("reseting vm");
```

File: device_driver.c [hunk 5]
```diff
@@ -339,63 +341,65 @@ static int parse_command(const char *data)
...
             {
-                pr_crit("reset_vms: %s", result.err);
+                pr_crit("could not reset vm # err[%s]", result.err);
                 status = -1;
                 goto cleanup;
             }
         }
         else if (strcmp("load_rules", command) == 0)
         {
+            pr_info("load rules");
```
Comment (baluchicken) [inline device_driver.c:356]:
```suggestion
            pr_debug("load rules");
```

File: device_driver.c [hunk 6]
```diff
@@ -339,63 +341,65 @@ static int parse_command(const char *data)
...
             }
 
-            pr_info("nasp: loading rules");
-
             load_opa_data(decoded);
             kfree(decoded);
         }
         else if (strcmp("load_config", command) == 0)
         {
+            pr_info("load config");
```
Comment (baluchicken) [inline device_driver.c:374]:
```suggestion
            pr_debug("load config");
```

File: device_driver.c [hunk 7]
```diff
@@ -339,63 +341,65 @@ static int parse_command(const char *data)
...
             {
-                pr_info("nasp: config load [%s]", decoded);
                 load_nasp_config(decoded);
-                pr_info("nasp: config loaded [%s]", decoded);
                 kfree(decoded);
             }
         }
         else if (strcmp("load_sd_info", command) == 0)
         {
+            pr_info("load sd info");
```
Comment (baluchicken) [inline device_driver.c:395]:
```suggestion
            pr_debug("load sd info");
```

File: device_driver.c [hunk 8]
```diff
@@ -462,7 +466,7 @@ static int parse_command(const char *data)
 /* Called when a process closes the device file. */
 static int device_release(struct inode *inode, struct file *file)
 {
-    pr_info("nasp: [%s] closed the communcation device", current->comm);
+    pr_info("communcation device closed # command[%s]", current->comm);
```
Comment (baluchicken) [inline device_driver.c:469]:
```suggestion
    pr_debug("communcation device closed # command[%s]", current->comm);
```

File: main.c [hunk 9]
```diff
@@ -34,7 +36,7 @@ MODULE_PARM_DESC(proxywasm_modules, "Enable/disable the proxywasm modules");
 
 static int __init nasp_init(void)
 {
-    pr_info("%s: module loaded at 0x%p running on %d CPUs", THIS_MODULE->name, nasp_init, nr_cpu_ids);
+    pr_info("module loaded at 0x%p running on %d CPUs", nasp_init, nr_cpu_ids);
```
Comment (baluchicken) [inline main.c:39]:
```suggestion
    pr_debug("module loaded at 0x%p running on %d CPUs", nasp_init, nr_cpu_ids);
```

File: opa.c [hunk 10]
```diff
@@ -241,7 +243,7 @@ static int parse_opa_builtins(opa_wrapper *opa, char *json)
             }
             else
             {
-                pr_warn("nasp: this opa module uses an unsupported builtin function: %s", name);
+                pr_warn("unsupported builtin function # name[%s]", name);
```
Comment (baluchicken) [inline opa.c:246]:
```suggestion
                pr_debug("unsupported builtin function # name[%s]", name);
```

File: opa.c [hunk 11]
```diff
@@ -614,12 +615,12 @@ opa_socket_context this_cpu_opa_socket_eval(const char *input)
 
     if (!opa)
     {
-        pr_warn("nasp: opa socket policy module not loaded, eval always evaluates to NULL");
+        pr_warn("opa socket policy module not loaded: eval always evaluates to NULL");
```
Comment (baluchicken) [inline opa.c:618]:
```suggestion
        pr_debug("opa socket policy module not loaded: eval always evaluates to NULL");
```

File: socket.c [hunk 12]
```diff
@@ -596,7 +597,7 @@ int nasp_recvmsg(struct sock *sock,
 	len = copy_to_iter(get_read_buffer(s), read_buffer_size, &msg->msg_iter);
 	if (len < read_buffer_size)
 	{
-		pr_warn("nasp: recvmsg copy_to_iter copied less than requested");
+		pr_warn("recvmsg copy_to_iter copied less than requested");
```
Comment (baluchicken) [inline socket.c:600]:
```suggestion
		pr_debug("recvmsg copy_to_iter copied less than requested");
```

File: socket.c [hunk 13]
```diff
@@ -709,14 +710,14 @@ static int configure_ktls_sock(nasp_socket *s)
 
 	if (params->cipher_suite != BR_TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256)
 	{
-		pr_warn("nasp: configure_ktls: only ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256 cipher suite is supported, got %x", params->cipher_suite);
+		pr_warn("configure kTLS error: only ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256 cipher suite is supported # requested_suite[%x]", params->cipher_suite);
```
Comment (baluchicken) [inline socket.c:713]:
```suggestion
		pr_debug("configure kTLS error: only ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256 cipher suite is supported # requested_suite[%x]", params->cipher_suite);
```

File: socket.c [hunk 14]
```diff
@@ -933,13 +934,13 @@ static int cache_and_validate_cert(nasp_socket *sc, char *key)
...
 	{
-		pr_err("nasp: found cert for id[%s]", key);
+		pr_debug("cert found # id[%s]", key);
 		sc->cert = cached_cert_bundle->cert;
 	}
 	// Validate the cached or the generated cert
 	if (!validate_cert(sc->cert->validity))
 	{
-		pr_warn("nasp: provided certificate is invalid");
+		pr_warn("remove invalid certificate from cache");
```
Comment (baluchicken) [inline socket.c:943]:
```suggestion
		pr_debug("remove invalid certificate from cache");
```

File: socket.c [hunk 15]
```diff
@@ -1483,11 +1482,11 @@ int socket_init(void)
 	u_int32_t result = generate_rsa_keys(rsa_priv, rsa_pub);
 	if (result == 0)
 	{
-		pr_err("nasp: socket_init error generating rsa keys");
+		pr_err("could not generate rsa keys");
 		return -1;
 	}
 
-	pr_info("nasp: socket support loaded.");
+	pr_info("socket support loaded");
```
Comment (baluchicken) [inline socket.c:1489]:
```suggestion
	pr_debug("socket support loaded");
```

File: socket.c [hunk 16]
```diff
@@ -1504,5 +1503,5 @@ void socket_exit(void)
 	free_rsa_private_key(rsa_priv);
 	free_rsa_public_key(rsa_pub);
 
-	pr_info("nasp: socket support unloaded.");
+	pr_info("socket support unloaded");
```
Comment (baluchicken) [inline socket.c:1506]:
```suggestion
	pr_debug("socket support unloaded");
```

File: wasm.c [hunk 17]
```diff
@@ -102,7 +104,7 @@ wasm_vm_result wasm_vm_new_per_cpu(void)
     int cpu = 0;
     for_each_possible_cpu(cpu)
     {
-        pr_info("nasp: creating vm for cpu %d", cpu);
+        pr_info("create vm # cpu[%d]", cpu);
```
Comment (baluchicken) [inline wasm.c:107]:
```suggestion
        pr_debug("create vm # cpu[%d]", cpu);
```

File: wasm.c [hunk 18]
```diff
@@ -115,7 +117,7 @@ wasm_vm_result wasm_vm_destroy_per_cpu(void)
         int cpu = 0;
         for_each_possible_cpu(cpu)
         {
-            pr_info("nasp: destroying vm for cpu %d", cpu);
+            pr_info("destroy vm # cpu[%d]", cpu);
```
Comment (baluchicken) [inline wasm.c:120]:
```suggestion
            pr_debug("destroy vm # cpu[%d]", cpu);
```

**Review by bonifaido — APPROVED** [review]

**Pull request merged**